PowerShell/update-deployment-settings.ps1

```diff
@@ -259,6 +259,48 @@
                 }
             }
 
+            $buildName = $buildDefinitionResult.name.ToLower()
```

Comment:
If we refactored the variable name and value as mentioned here https://github.com/microsoft/coe-starter-kit/pull/3402#discussion_r942492011. This could be simplified somewhat to something like the following.

```
           # Declare Solution Upgrade Tags; By default is 'update'
            $solutionUpgradeInputValue = $false
            $pipelineVariableName =  "TriggerSolutionUpgrade"
            if($null -ne $configurationData) 
            {
                foreach($deploymentEnvironment in $configurationData) 
                {
                    $deploymentBuildName = $deploymentEnvironment.BuildName.ToLower()
                    Write-Host "deploymentBuildName - " $deploymentBuildName
                    if($deploymentBuildName -eq $buildName)
                    {
                        $solutionUpgradeTag = $deploymentEnvironment.UserSettings | Where-Object { $_.Name -eq $pipelineVariableName}
                        if($null -ne $solutionUpgradeTag -and $null -ne $solutionUpgradeTag.Value)
                        {
                            $solutionUpgradeInputValue = $solutionUpgradeTag.Value

                            Write-Host "solutionUpgradeInputValue for  $deploymentBuildName - $solutionUpgradeInputValue"

                            $found = $false
                            #See if the variable already exists
                        $buildVariable = $newBuildDefinitionVariables.PSObject.Properties | Where-Object { $_.Name -eq $pipelineVariableName}
                        if($null -eq $buildVariable -or $null -eq $buildVariable .Value) {
                                $newBuildDefinitionVariables | Add-Member -MemberType NoteProperty -Name $pipelineVariableName -Value @{value = ''}
                            }
                            # Set the value to the value passed in on the configuration data
                            $newBuildDefinitionVariables.$pipelineVariableName.value = $solutionUpgradeInputValue
                        }
                     break;
                    }
                }
            }
```

Reply:
Done. Refactored code as specified.